lib/src/connection.dart

```diff
@@ -192,6 +203,19 @@ class PostgreSQLConnection extends Object
   /// After the returned [Future] completes, this connection can no longer be used to execute queries. Any queries in progress or queued are cancelled.
   Future close() => _close();
 
+  /// Adds a Client Message to the existing connection
+  ///
+  /// This is a low level API and the message must follow the protocol of this
+  /// connection. It's the responsiblity of the caller to ensure this message
+  /// does not interfere with any running queries or transactions.
+  void addMessage(ClientMessage message) {
```

Comment:
I'm not sure if this should be on the public API. Just to be sure, please annotate with https://pub.dev/documentation/meta/latest/meta/internal-constant.html

Reply:
I thought this would be fine since `socket` is already publicly exposed without the internal annotation. 

The user can still add messages using `connection.socket!.add(message.asBytes())`. 

Also, when using Streaming Replication Protocol, the user needs a way to respond to messages. Marking it as `internal` would make the analyzer complain without providing an alternative.

Reply:
Oh, I misunderstood this part.

The `socket` being public is something I would like to change in the future, reduce the public API so that clients can't really screw up the connection internals.

However, I see now why these should be public. Let it be then, for a future v3 with breaking changes we can redesign this a bit.